app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -62,9 +69,28 @@ class MainActivity : AppCompatActivity() {
                 requestWeather(intent.getStringExtra("cityName").toString())
             }
         }
+        airplaneModeReceiver = object : BroadcastReceiver() {
+            override fun onReceive(context: Context, intent: Intent) {
+                val isAirplaneModeEnabled = intent.getBooleanExtra("state", false)
+                if (isAirplaneModeEnabled) {
+                    binding.buttonGetWeather.setText(R.string.airplane_on)
+                    binding.buttonGetWeather.isEnabled = false
+                } else {
+                    binding.buttonGetWeather.setText(R.string.get_weather_button)
+                    binding.buttonGetWeather.isEnabled = true
```

Comment:
instead of this line, use
`binding.buttonGetWeather.isEnabled = isAirplaneModeEnabled`
outside of `if-else` construct.

Reply:
or even better:
do not manipulate button parameters manually if you already have an  existing function `enableGetWeatherButton`. Create new function `disableGetWeatherButton` for disabling it.

Reply:
I have created new function `disableGetWeatherButton` and added `binding.buttonGetWeather.setText(R.string.get_weather_button);
        binding.buttonGetWeather.isEnabled = true` to `enableGetWeatherButton()`